### PR TITLE
Added Stack in Typescript

### PR DESCRIPTION
**Pull request by DanielRomeo**
**Fixes #{#643}**

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [x] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [x] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [x] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [x] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [x] This pull request shall only be merged once the build passes.

-   [x] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

@Danushka96

### CODE REVIEW

File: data-structures/ds-stk/typescript/stack.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    // returns the number of items in the stack:
+    public size = (): number => {
+        return this.count;
+    }
+
+    //returns the stack list:
+    public showFullStack = () => {
+        return this.storage;
+    }
+};
```
Comment (RavinduSachintha) [inline data-structures/ds-stk/typescript/stack.ts:37]: Codacy found an issue: [Unnecessary semicolon.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6362486)

File: data-structures/ds-stk/typescript/stack.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    public push = <T extends unknown>(item: T ): void  => {
+        this.storage[this.count] = item;
+        this.count++;
+
+    };
+
+    public pop = () : objected => {
+        if(this.count == 0){
+            return undefined;
```
Comment (RavinduSachintha) [inline data-structures/ds-stk/typescript/stack.ts:15]: Codacy found an issue: [Unexpected use of undefined.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6362486)

File: data-structures/ds-stk/typescript/stack.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    private storage: any = {};
+
+    public push = <T extends unknown>(item: T ): void  => {
+        this.storage[this.count] = item;
+        this.count++;
+
+    };
+
+    public pop = () : objected => {
+        if(this.count == 0){
```
Comment (RavinduSachintha) [inline data-structures/ds-stk/typescript/stack.ts:14]: Codacy found an issue: [Expected '===' and instead saw '=='.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6362486)

File: algorithms/ar-isrt/typescript/insertion_sort.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+const swap = (arr: Array<number>, a: number, b: number) => {
```
Comment (Danushka96) [inline algorithms/ar-isrt/typescript/insertion_sort.ts:1]: Hi @DanielRomeo 
Remove this file, this is not acceptable,

**Review by Danushka96 — CHANGES_REQUESTED** [review]

File: algorithms/ar-isrt/typescript/insertion_sort.ts (hunk 4 above)
Comment (DanielRomeo) [reply algorithms/ar-isrt/typescript/insertion_sort.ts:1]: hello @Danushka96  what did i do wrong?

Comment (Danushka96) [reply algorithms/ar-isrt/typescript/insertion_sort.ts:1]: @DanielRomeo This PR is about stack right? why there's an insertion sort in the commits?

Comment (DanielRomeo) [reply algorithms/ar-isrt/typescript/insertion_sort.ts:1]: oh. I have no idea, my insertion sort pull request was accepted a few hours ago, i don't know why my commit that got added again to this PR. Now I'm confused, what exactly should I do? should I delete this PR entirely and create a new one? or?

Comment (DanielRomeo) [reply algorithms/ar-isrt/typescript/insertion_sort.ts:1]: @Danushka96

Comment (Danushka96) [reply algorithms/ar-isrt/typescript/insertion_sort.ts:1]: Yes, let's make a clear PR,

**Pull request closed without merging**